src/transport.rs

```diff
@@ -190,7 +196,10 @@ where
             }
         };
 
-        join!(future, print_output, process_server_responses, message_fut);
+        tokio::select! {
+            (..) = async { join!(future, print_output, process_server_responses, futures::future::pending::<()>()) } => unreachable!(),
+            () = message_fut => {},
+        };
```

Comment:
Why is there `futures::fturue::pending` here?

Reply:
Ah, it's to keept that branch alive. I think maybe we should just join on all the futures because they should cleanly exit each other.

Reply:
Or I guess this is necessary to ensure that an exit notification doesn't cause other pending things to still be sent? Maybe add a comment?

Reply:
I'm not sure `future` (stdin reader) would exit if it's waiting for input. Will leave a comment